Cargo.toml

```diff
@@ -76,6 +76,9 @@ members = [
     "network/discovery",
     "network/memsocket",
     "network/netcore",
+    "network/network-service/client",
+    "network/network-service/server",
+    "network/network-service/types",
```

Comment:
Can we stick with 1 crate?  What's the reasoning for breaking it down into 3 different crates?  I'm not sure binary size is particularly important at this point.

Reply:
I think it's cleaner than one giant crate, especially as we're thinking about microservices going forward and alternate implementations. Splitting things up like this makes it much easier to reason about (clean client/server separation) and will make future tooling/implementations cleaner too. I'm inclined to leave it like this. Note: it also matches the storage service layout.

Reply:
I'm of the opinion microservices solve two problems.

1. The problem of too large of a team, and therefore breaking down into smaller chunks for different people to focus on.
2. Throughout horizontal scale at the cost of latency and overhead, by being able to deploy everything in separate containers or machines.

I don't think either of those are problems today.  We can go forward with this, I just want to make sure you keep it simple so:
1. You can move fast and not deal with semantic problems of what belongs where
2. Reviewers don't have to jump between multiple places to determine what's going on.

Reply:
Noted 😄 I feel a bit differently, but given that it's relatively easy to just merge everything into a single crate in the future, let's proceed as-is.